### DISCUSSION TITLE
Is there any way with kcadm.sh to create an organization and add members in it?

### DISCUSSION THREAD

**Discussion by bdovaz** [keycloak/keycloak#35572] [Q&A] [answered]
I see that it is not reflected in the documentation and I do not know if it is not possible yet or if it is not documented:

https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/admin-cli.html

To create the organization I see that it is like this:

```bash
./kcadm.sh create organizations -r myRealm -s alias=myAlias -s name=myName -s domains[0]=myDomain
```

But to add a member even though I think I'm doing it right, it doesn't find the user and I know it exists:

```bash
./kcadm.sh create "organizations/{ORGANIZATION_ID}/members" -r myRealm -s id=myUserId
```

I get: `User does not exist`

https://github.com/keycloak/keycloak/blob/364572046f6d52fe34ab038686786d4135da1ceb/services/src/main/java/org/keycloak/organization/admin/resource/OrganizationMemberResource.java#L91

**Comment by nicolashery** [comment, marked as answer]
I also ran into this. Got it to work by sending the user ID as the whole request body (`-b`), instead of JSON (`-s id=...`). That endpoint is a little odd it says it accepts JSON but really it's just plain text with just the user ID. See https://github.com/keycloak/keycloak/issues/34401

```
$ kcadm.sh create organizations/b3e797be-3c33-4720-8ece-bb9e6816eeda/members -r example -b 904b6bfa-3bbf-46e7-9fa6-6e10f9be7563

Created new member with id '904b6bfa-3bbf-46e7-9fa6-6e10f9be7563'
```
